src/daemon/auth/providers/openid_connect/provider.rs

```diff
@@ -357,49 +363,84 @@ impl OpenIDConnectAuthProvider {
+                                                    r.to_string()
+                                                )));
+                                            }
+                                            openidconnect::RequestTokenError::Parse(r, _) => {
+                                                return Err(CoreErrorResponseType::Extension(format!(
+                                                    "Error while parsing new token: {}",
+                                                    r.to_string()
+                                                )));
+                                            }
+                                            openidconnect::RequestTokenError::Other(err_string) => {
+                                                match err_string.as_str() {
+                                                    "temporarily_unavailable" | "server_error" => {
+                                                        return Err(CoreErrorResponseType::Extension(
+                                                            err_string.to_string(),
+                                                        ))
+                                                    }
+                                                    _ => {
+                                                        return Err(CoreErrorResponseType::Extension(format!(
+                                                            "Unknown Error(#1) while receiving new token: {}",
+                                                            err_string
+                                                        )))
+                                                    }
+                                                }
+                                            }
+                                        }
                                     }
                                 }
                             }
                             None => {
                                 // should be unreachable
-                                warn!(
-                                    "OpenID Connect: Could not extend login session for user '{}': {}",
-                                    &session.id, "Connection to OpenID Connect provider not yet established"
-                                );
+                                return Err(CoreErrorResponseType::Extension(
+                                    "Connection to OpenID Connect provider not yet established".to_string(),
+                                ));
                             }
                         }
                     }
                     Err(err) => {
-                        warn!(
-                            "OpenID Connect: Could not extend login session for user '{}': {}",
-                            &session.id, err
-                        );
+                        return Err(CoreErrorResponseType::Extension(
```

Comment:
Suggestion: Prefix the error message with `krill_internal_error: ` to make it easy to distinguish from any actual provider error messages. Also make Error lowercase and remove the #2 (see comment above regarding #1).